core/ecschema-editing/src/Merging/PropertyMerger.ts

```diff
@@ -115,20 +114,20 @@ async function modifyClassProperty(context: SchemaMergeContext, itemKey: SchemaI
   }
 
   if (propertyProps.description !== undefined) {
-    property.setDescription(propertyProps.description);
+    await context.editor.entities.properties.setDescription(itemKey, property.name, propertyProps.description);
   }
   if (propertyProps.label !== undefined) {
-    property.setLabel(propertyProps.label);
+    await context.editor.entities.properties.setLabel(itemKey, property.name, propertyProps.label);
   }
   if (propertyProps.isReadOnly !== undefined) {
-    property.setIsReadOnly(propertyProps.isReadOnly);
+    await context.editor.entities.properties.setIsReadOnly(itemKey, property.name, propertyProps.isReadOnly);
   }
   if (propertyProps.priority !== undefined) {
-    property.setPriority(propertyProps.priority);
+    await context.editor.entities.properties.setPriority(itemKey, property.name, propertyProps.priority);
   }
 
   if (property.isArray()) {
-    await arrayProperty.merge(property as any, propertyProps);
+    await arrayProperty.merge(context, itemKey, property as any, propertyProps);
```

Comment:
Looks like i've missed an any-cast, lets better cast to property to MutableArrayProperty instead.